state.go

```diff
@@ -317,6 +292,66 @@ func (m *Memberlist) probeNode(node *nodeState) {
 	m.suspectNode(&s)
 }
 
+// Ping initiates a ping to the node with the specified name.
+func (m *Memberlist) Ping(node string) (bool, time.Duration, error) {
```

Comment:
Sounds like the best way to perform this type of error checking is to define a new error type that describes the case in which a ping was successfully transmitted but no response was received.  I'll work on this change and add it to the pull request once it's complete.  Thanks for the help!